include/envoy/http/async_client.h

```diff
@@ -30,11 +43,42 @@ class AsyncClient {
 
   /**
    * Notifies caller of async HTTP request status.
+   *
+   * To support a use case where a caller makes multiple requests in parallel,
+   * individual callback methods provide request context corresponding to that response.
    */
   class Callbacks {
   public:
     virtual ~Callbacks() = default;
 
+    /**
+     * Called when the async HTTP request succeeds.
+     * @param request  request handle or nullptr if no request could be created.
+     *                 NOTE: request handle is passed for correlation purposes only, e.g.
+     *                 for client code to be able to exclude that handle from a list of
+     *                 requests in progress.
+     * @param response the HTTP response
+     */
+    virtual void onRequestSuccess(const Request* request, ResponseMessagePtr&& response) PURE;
+
+    /**
+     * Called when the async HTTP request fails.
+     * @param request request handle or nullptr if no request could be created.
+     *                NOTE: request handle is passed for correlation purposes only, e.g.
+     *                for client code to be able to exclude that handle from a list of
+     *                requests in progress.
+     * @param reason  failure reason
+     */
+    virtual void onRequestFailure(const Request* request, FailureReason reason) PURE;
+  };
+
+  /**
+   * Notifies caller of async HTTP request status.
+   */
+  class RequestCallbacks : public Callbacks {
```

Comment:
I'm not sure if having two variations of this interface is worth it, would it be possible to just update the existing implementations to override the function with the request ptr parameter?

Reply:
`RequestCallbacks` has been removed